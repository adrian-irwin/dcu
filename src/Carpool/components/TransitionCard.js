import React, { useEffect, useState } from "react";
import { Platform } from "react-native";
import { Card, Text } from "react-native-paper";

const TransitionCard = ({ title, description, navigation }) => {
    const [image, setImage] = useState(null);

    // Function to select the correct image for the card based on the title
    const imageSelector = () => {
        if (title === "Driver") {
            setImage(require("../assets/driver.png"));
        }
        else if (title === "Passenger") {
            setImage(require("../assets/passenger.png"));
        }
    };

    // Call the imageSelector function when the title changes
    useEffect(() => {
        imageSelector();
    }, [title]);

    return (
        <Card
            style={{
                marginVertical: 10,
                shadowColor: "#000",
                shadowOffset: { width: 1, height: 5 },
                shadowOpacity: 0.5
            }}
            onPress={() => {
                console.log(`user on ${Platform.OS} pressed ${title} card`);
                navigation.navigate(description);
            }}>
            <Card.Content style={{ marginBottom: 15 }}>
                <Text variant="displayLarge" style={{ textAlign: "center" }} >{title}</Text>
            </Card.Content>
            <Card.Cover source={image} />
        </Card>
    );
};

export default TransitionCard;
