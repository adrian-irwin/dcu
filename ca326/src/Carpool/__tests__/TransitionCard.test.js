import React from "react";
import { render } from "@testing-library/react-native";

import TransitionCard from "../components/TransitionCard";

describe("TransitionCard", () => {
    it("renders correctly", () => {
        const { toJSON } = render(<TransitionCard />);
        expect(toJSON()).toMatchSnapshot();
    });
});
