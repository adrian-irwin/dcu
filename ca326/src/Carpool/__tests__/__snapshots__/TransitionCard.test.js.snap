// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransitionCard renders correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(247, 243, 249, 1)",
          "borderRadius": 12,
          "elevation": 1,
          "marginVertical": 10,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 5,
            "width": 1,
          },
          "shadowOpacity": 0.5,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexShrink": 1,
          }
        }
        testID="card"
      >
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "paddingTop": 16,
              },
              Object {
                "marginBottom": 15,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 57,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 64,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "textAlign": "center",
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#eeeeee",
                "height": 195,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 12,
              },
              undefined,
            ]
          }
        >
          <Image
            accessibilityIgnoresInvertColors={true}
            siblings={
              Array [
                "Card.Content",
                "withTheme(Card.Cover)",
              ]
            }
            source={null}
            style={
              Array [
                Object {
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "flex-end",
                  "padding": 16,
                  "width": undefined,
                },
                Object {
                  "borderRadius": 12,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
